pkg/nvcdi/driver-nvml.go

```diff
@@ -102,17 +102,17 @@ func (l *nvcdilib) NewDriverLibraryDiscoverer(version string) (discover.Discover
 	driverDotSoSymlinksDiscoverer := discover.WithDriverDotSoSymlinks(
 		libraries,
 		version,
-		l.nvidiaCDIHookPath,
+		l.hookCreator,
 	)
 	discoverers = append(discoverers, driverDotSoSymlinksDiscoverer)
 
-	if l.HookIsSupported(HookEnableCudaCompat) {
-		// TODO: The following should use the version directly.
-		cudaCompatLibHookDiscoverer := discover.NewCUDACompatHookDiscoverer(l.logger, l.nvidiaCDIHookPath, l.driver)
-		discoverers = append(discoverers, cudaCompatLibHookDiscoverer)
-	}
+	// TODO: The following should use the version directly.
```

Comment:
Question: if we're just refactoring the hook creation, should we not still check `HookIsSupported` here?

Reply:
Yes, I'm not saying we should keep the function. I'm saying that for the first refactoring step, we ONLY add a `HookCreator` and then as a follow-up we pull in the ability to disable hooks.

Reply:
Done

Reply:
Please update this diff so that the only modification is the chane from `l.nvidiaCDIHookPath` -> `l.hookCreator`.

Reply:
done